lint-tasks.yml

```diff
@@ -0,0 +1,217 @@
+version: "3"
+
+vars:
+  LINT_VENV_DIR: "{{.BUILD_DIR}}/lint-venv"
+  LINT_VENV_CHECKSUM_FILE: "{{.BUILD_DIR}}/lint#venv.md5"
+  # Linter target dirs
+  PYTHON_LINT_DIRS: ["{{.ROOT_DIR}}/clp_ffi_py", "{{.ROOT_DIR}}/tests"]
+  CPP_LINT_DIRS: ["{{.CLP_FFI_PY_SRC_DIR}}"]
+
+tasks:
+  check:
+    cmds:
+      - task: "cmake-check"
+      - task: "cpp-check"
+      - task: "py-check"
+      - task: "yml-check"
+
+  fix:
+    cmds:
+      - task: "cmake-fix"
+      - task: "cpp-fix"
+      - task: "py-fix"
+      - task: "yml-fix"
+
+  cmake-check:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--check"
+
+  cmake-fix:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--in-place"
+
+  cpp-configs:
+    # TODO: remove this once all clang-tidy has been fixed
```

Comment:
Depends on how you look at the dependencies. 

One argument is that: config-cmake-project is only needed by static check, so adding it as the dependency of cpp-static-check and cpp-static-fix should be good enough. so we don't need it to be a dependency of cpp-configs

You can also argue that you want the compilation database to be generated without having to run linter-fix, then maybe we should simply expose config-cmake-project to user (currently it's internal). The trade off is that now you need to manually run "task config-cmake-project" to generate the compilation database.

Imo, having it as a dependency of cpp-configs is a bit of hacky (and is a temporary solution), because cpp-configs is supposed to run only once to generate the symlink, but the compliation database should be generated whenever cmake file changes.

Reply:
kk I see ur point. Make sense. Let's get back to this after clang-tidy fix